### PR TITLE
New York State Working Families Tax Credit reform

### LINKED ISSUE #4650: Parameters for New York State Working Families Tax Credit reform
Proposed in [S277B](https://legislation.nysenate.gov/pdf/bills/2023/S277B)

Replaces the NYS EITC, CTC, and dependent exemption with a single credit for each child under 18, phasing out over an income range but not below a minimum (so can't be modeled with current basic income parameters). Also preserves the dependent exemption for dependents age 18 or older.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4650

### CODE REVIEW

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  unit: currency-USD
+  period: year
+  uprating:
+    parameter: gov.irs.uprating
+    rounding:
+      type: downwards
+      interval: 50
+  label: New York Working Families Tax Credit amount
+  reference: 
+    - title: Instructions for Mississippi State FORMS 80-105 AND 80-205 (2)(A)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml:12]: hmm

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/eitc/match.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/eitc/match.yaml:1]: can we use the existing parameter for this? doesn't look like structurally altered the ny eitc

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml:12]: Will even out the formatting in a bit, still missing some rules

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/eitc/match.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/contrib/states/ny/wftc/eitc/match.yaml:1]: It seems easier to represent that with new parameters

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+description: New York provides the following maximum Working Families Tax Credit amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml:1]: Same for minimum IIUC
```suggestion
description: New York provides the following maximum Working Families Tax Credit amount per child.
```

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml:1]: This is only defined once in the legal code, can we unify the parameter with `dependent_exemption/child_age_threshold`? e.g. `wftc/child_age_threshold.yaml`?

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/eitc/match.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+description: New York matches this percentage of the Earned Income Tax Credit under the Working Families Tax Credit.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/eitc/match.yaml:1]:
```suggestion
description: New York matches this fraction of the Earned Income Tax Credit under the Working Families Tax Credit.
```

**Review by MaxGhenis — APPROVED** [review]
Good to merge after these minor suggestions

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml:1]: This is defined twice, one time for each the exemption and the WFTC

**Comment by PavelMakarchuk** [comment]
> Good to merge after these minor suggestions

Added the EITC split

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml:1]: Could you share where? I only see one `eighteen` in the law:

>   (C) "QUALIFYING CHILD" OR "QUALIFYING CHILDREN" SHALL MEAN AS  DEFINED
 IN  26 USC §24(C)(1) EXCEPT THAT SUCH TERM SHALL ALSO INCLUDE QUALIFYING
 CHILDREN WHO HAVE NOT ATTAINED THE AGE OF EIGHTEEN.

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml:1]: Adjusted! You were right

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+description: New York limits the Working Families Tax Credit to children below this age threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/child_age_threshold.yaml:1]:
```suggestion
description: New York limits the Working Families Tax Credit to children below this age.
```

File: policyengine_us/tests/policy/contrib/states/ny/ny_working_families_tax_credit.yaml [hunk 5]
```diff
@@ -0,0 +1,211 @@
+- name: Reform does not apply
+  period: 2024
+  # reforms: policyengine_us.reforms.states.ny.wftc.ny_working_families_tax_credit.ny_working_families_tax_credit
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/contrib/states/ny/ny_working_families_tax_credit.yaml:3]: remove

**Review by MaxGhenis — APPROVED** [review]
yikes! but ok.

**Pull request merged**